cmsis-core-nrf51822/compiler_abstraction.h

```diff
@@ -35,50 +37,63 @@
 #if defined ( __CC_ARM )
     
     #ifndef __ASM
-        #define __ASM               __asm                       /*!< asm keyword for ARM Compiler */
+        #define __ASM               __asm                       
```

Comment:
@rgrover, @LiyouZhou: There is a small issue with this file because it defines macros such as __ASM, __INLINE, etc that are also defined [here](https://github.com/ARMmbed/cmsis-core/blob/master/cmsis-core/core_cm0.h). The problem is that the second definition does not include an `#ifndef` and the compiler throws a lot of warning because of this re-definition. Any ideas on how to fix this?